src/tests/roc_audio/test_profiler.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <CppUTest/TestHarness.h>
+
+#include "roc_audio/profiler.h"
+#include "roc_core/heap_allocator.h"
+
+namespace roc {
+namespace audio {
+
+namespace {
+
+struct TestFrame {
+    TestFrame(size_t size, core::nanoseconds_t time)
+        : size_(size)
+        , time_(time) {
+    }
+
+    size_t size_;
+    core::nanoseconds_t time_;
+};
+
+const core::nanoseconds_t millisecond = 1000000L;
+const core::nanoseconds_t interval = 50 * millisecond; // 5 chunks
+const int sample_rate = 5000;                          // 50 samples / chunk
+const int num_channels = 1;
+core::HeapAllocator allocator;
+
+} // namespace
+
+TEST_GROUP(profiler) {};
+
+TEST(profiler, test_moving_average) {
+    Profiler profiler(allocator, num_channels, sample_rate, interval);
+
+    TestFrame frames[9] = {
```

Comment:
Nitpick: we can omit `9` here and in expected_moving_average. And we can use ROC_ARRAY_SIZE in the for loop. Also I'd add an assertion that the size of the two arrays is the same.